ouroboros-consensus-cardano/src/shelley/Ouroboros/Consensus/Shelley/Ledger/Query.hs

```diff
@@ -297,6 +298,15 @@ data instance BlockQuery (ShelleyBlock proto era) :: Type -> Type where
   -- longer used (because mainnet has hard-forked to a newer version), it can be
   -- removed.
 
+  GetAccountState 
+    :: ( CG.ConwayEraGov era
+       , LC.EraTxOut era
+       , Show (StashedAVVMAddresses era)
+       , Eq (StashedAVVMAddresses era)
+       )
+    => SL.NewEpochState era
```

Comment:
Does this query expect a full NewEpochState? This seems wrong. There is an `st` parameter when answering the queries that is the ledgerstate at the requested point (tip or immutable tip) so this state shall not be an argument to the query. Otherwise we would be sending whole ledgerstates between clients and nodes.